src/wallet/rpcwallet.cpp

```diff
@@ -40,17 +40,60 @@
 #include "sc/sidechainrpc.h"
 #include "consensus/validation.h"
 
+#include "zen/forks/fork11_shieldedpooldeprecationfork.h"
+
 using namespace std;
 
 using namespace libzcash;
 using namespace Sidechain;
 
 namespace{
-
     int GetJoinSplitSize(int shieldedTxVersion) {
         return JSDescription::getNewInstance(shieldedTxVersion == GROTH_TX_VERSION).GetSerializeSize(SER_NETWORK, PROTOCOL_VERSION, shieldedTxVersion);
+    }    
+}
+
+// anonymous namespace related to shielded pool deprecation hard fork
+// TODO: REMOVE ONCE SHIELDED POOL DEPRECATION HARD FORK IS "HIGHLY FINAL"
+namespace{
+    std::string shieldedPoolDeprecationErrorMessage = "This method is deprecated due to shielded pool deprecation hard fork.";
```

Comment:
Just a small optimization, feel free to ignore this, but here you do not need to init a string.
```suggestion
    const char* shieldedPoolDeprecationErrorMessage = "This method is deprecated due to shielded pool deprecation hard fork.";
```

or even

```suggestion
    constexpr std::string_view shieldedPoolDeprecationErrorMessage = "This method is deprecated due to shielded pool deprecation hard fork.";
```

Reply:
I'd go with first option which doesn't require additional modifications; addressed in ca71777097aece83e8f83375243baeac51c0451d.